futures-executor/src/current_thread.rs

```diff
@@ -122,12 +122,12 @@ impl<'a> Context<'a> {
     /// only be acquired from the `run` function in this module, which means
     /// that other futures may be spawned on the current thread as a daemon or
     /// not.
-    pub fn block_on<F>(&mut self, future: F) -> Result<F::Item, F::Error>
+    pub fn block_on<F>(&mut self, mut future: F) -> Result<F::Item, F::Error>
         where F: Future,
     {
-        let mut future = task::spawn(future);
+        let mut map = LocalMap::new();
         loop {
-            match future.poll_future_notify(&::IntoNotifyHandle(self.handle), 0) {
+            match future.poll(&mut task::Context::new(&mut map, 0, self.handle)) {
```

Comment:
IMO this method is a bit confusing-- if i call `task::Context::new(&mut map, 0, handle).block_on(fut)` manually, it seems like i could reasonably expect it to use the `LocalMap` I provided, rather than generating a new one for this specific task. I guess this function will probably go away (or at least look different) in the new executor design, so maybe it's not worth worrying about now. What do you think?

Reply:
Yeah I think let's revisit then.